pkg/controller/utils/utils.go

```diff
@@ -138,6 +140,67 @@ func TryUpdate(ctx context.Context, backoff wait.Backoff, c client.Client, obj r
 	return tryUpdate(ctx, backoff, c, obj, c.Update, transform)
 }
 
+// TypedCreateOrUpdate is like controllerutil.CreateOrUpdate, it retrieves the current state of the object from the
+// API server, applies the given mutate func and creates or updates it afterwards. In contrast to
+// controllerutil.CreateOrUpdate it tries to create a new typed object of obj's kind (using the provided scheme)
+// to make typed Get requests in order to leverage the client's cache.
+func TypedCreateOrUpdate(ctx context.Context, scheme *runtime.Scheme, c client.Client, obj *unstructured.Unstructured, mutate func() error) error {
```

Comment:
nice :) Does a unit test make sense for this function?

Reply:
Yes, done 👍